--- ../tools/maillogconvert.pl	2014-01-29 16:09:00.000000000 +0800
+++ maillogconvert.pl	2015-03-27 11:07:58.883606314 +0800
@@ -126,10 +126,11 @@
 	my $size=shift||0;
 	my $forwardto=shift;
 	my $extinfo=shift||'-';
+	my $subject_s=shift||'-';
 
 	# Clean day and month
 	$day=sprintf("%02d",$day);
-    $month=sprintf("%02d",$MonthNum{$month}||$month);
+	$month=sprintf("%02d",$MonthNum{$month}||$month);
 
 	# Clean from
 	$from=&CleanEmail($from);
@@ -154,7 +155,7 @@
 	#if we don't have info for relay_s, we keep it unknown, awstats might then guess it
 	
 	# Write line
-	print "$year-$month-$day $time $from $to $relay_s $relay_r SMTP $extinfo $code $size\n";
+	print "$year-$month-$day $time $from $to $relay_s $relay_r SMTP $extinfo $code $size \"Subject=$subject_s\"\n";
 	
 	# If there was a redirect
 	if ($forwardto) {
@@ -598,7 +599,18 @@
 			$mail{$id}{'extinfo'}=~s/\s/_/g;
 			$mailid=$id;
 		}
-	}
+	} 
+	# 
+	# Mod for log down header
+	#
+	elsif (/: info: header Subject:/) {
+		# Example: "Mar 24 15:40:26 jecdevvm-netbd01 postfix/smtp[27848]: D7576921F2: info: header Subject: X'mas Purchasing"
+		$MailType||='postfix';
+                my ($id,$subject_s)=m/\w+\s+\d+\s+\d+:\d+:\d+\s+[\w\-\.\@]+\s+(?:sendmail|postfix\/(?:local|lmtp|smtpd|smtp|virtual|pipe))\[\d+\]:\s+(.*?):\s+(?:info: header Subject:)\s+(.*)/;
+		$mailid=$id;
+		$mail{$mailid}{'subject_s'}=$subject_s;
+		debug("For id=$mailid, found a postfix info header: \"$mail{$mailid}{'subject_s'}\"");
+        }
 	
 	
 	#
@@ -609,7 +621,7 @@
 		my $delivery=0;
 		my $canoutput=0;
 		
-		debug("ID:$mailid RELAY_S:".($mail{$mailid}{'relay_s'}||'')." RELAY_R:".($mail{$mailid}{'relay_r'}||'')." FROM:".($mail{$mailid}{'from'}||'')." TO:".($mail{$mailid}{'to'}||'')." CODE:".($mail{$mailid}{'code'}||''));
+		debug("ID:$mailid RELAY_S:".($mail{$mailid}{'relay_s'}||'')." RELAY_R:".($mail{$mailid}{'relay_r'}||'')." FROM:".($mail{$mailid}{'from'}||'')." TO:".($mail{$mailid}{'to'}||'')." CODE:".($mail{$mailid}{'code'}||'')." SUBJECT:".($mail{$mailid}{'subject_s'}||''));
 
 		# Check if we can output a mail line
 		if ($MailType eq 'qmail') {
@@ -638,7 +650,7 @@
 
 		# If we can
 		if ($canoutput) {
-			&OutputRecord($mail{$mailid}{'year'}?$mail{$mailid}{'year'}:$year,$mail{$mailid}{'mon'},$mail{$mailid}{'day'},$mail{$mailid}{'time'},$mail{$mailid}{'from'},$to,$mail{$mailid}{'relay_s'},$mail{$mailid}{'relay_r'},$code,$mail{$mailid}{'size'},$mail{$mailid}{'forwardto'},$mail{$mailid}{'extinfo'});
+			&OutputRecord($mail{$mailid}{'year'}?$mail{$mailid}{'year'}:$year,$mail{$mailid}{'mon'},$mail{$mailid}{'day'},$mail{$mailid}{'time'},$mail{$mailid}{'from'},$to,$mail{$mailid}{'relay_s'},$mail{$mailid}{'relay_r'},$code,$mail{$mailid}{'size'},$mail{$mailid}{'forwardto'},$mail{$mailid}{'extinfo'},$mail{$mailid}{'subject_s'});
 			# Delete mail with generic unknown id (This id can by used by another mail)
 			if ($mailid eq '999') {
 				debug(" Delete mail for id=$mailid",3);
